### PR TITLE
[BugFix] fix deprecated CURL type

### PR DESCRIPTION
**Pull request by murphyatwork**
Why I'm doing:

`CURLINFO_CONTENT_LENGTH_DOWNLOAD` is deprecated since curl 7.55.0.

What I'm doing:

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by imay — DISMISSED** [review]

**Comment by imay** [comment]
> CURLINFO_CONTENT_LENGTH_DOWNLOAD is deprecated since curl 7.55.0.

If you can provide the link in the commit message, it will be better.

**Review by satanson — DISMISSED** [review]

**Review by ZiheLiu — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by kevincai** [comment]
missing the branch-3.3 backport

File: be/src/http/http_client.h [hunk 1]
```diff
@@ -93,7 +93,11 @@ class HttpClient {
     // used to get content length
     int64_t get_content_length() const {
         double cl = 0.0f;
+#if (CURLVERSION_NOW >= CURLVERSION_TENTH)
```
Comment (kevincai) [inline be/src/http/http_client.h:96]: we can safely use the new name since our curl upgraded to v8.4.0

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by kevincai** [comment]
done in #49351

**Pull request closed without merging**